decidim-admin/app/commands/decidim/admin/bulk_action.rb

```diff
@@ -0,0 +1,91 @@
+# frozen_string_literal: true
+
+module Decidim
+  module Admin
+    class BulkAction < Decidim::Command
+      # Public: Initializes the command.
+
+      def initialize(user, action, reportables)
+        @user = user
+        @action = action
+        @reportables = reportables
+        @result = { ok: [], ko: [] }
+      end
+
+      # Executes the command. Broadcasts these events:
+      #
+      # - :ok when everything is valid, together with the resource.
+      # - :invalid if the resource is not reported
+      #
+      # Returns nothing.
+      def call
+        return broadcast(:invalid) if reportables.blank?
+
+        bulk_action!
+        create_action_log if user
+
+        broadcast(:ok, **result)
+      end
+
+      private
+
+      attr_reader :action, :reportables, :user, :result
+
+      def create_action_log
+        action_log_type = case action
+                          when "hide"
+                            "bulk_hide"
+                          when "unreport"
+                            "bulk_unreport"
+                          when "unhide"
+                            "bulk_unhide"
+                          end
+
+        Decidim::ActionLogger.log(
+          action_log_type,
+          user,
+          user,
+          nil,
+          extra: {
+            reported_content:,
+            reportable_type: reported_content.values
+          }
+        )
+      end
+
+      def reported_content
+        @reported_content ||= result[:ok].to_h { |reportable| [reportable.id, reportable.class.name.split("::").last] }
```

Comment:
```suggestion
        @reported_content ||= result[:ok].to_h { |moderation| [moderation.reportable.id, moderation.title] }
```
Let's not do this, it is not translated and gives no info about the proposal.
Let's just add the reportable id and in the presenter we deal with it

In the suggestion, we should deal with moderations, not reportables. This will force us to add the `title` method to the moderation class, that could be like this:

```ruby
   
  def title
    return reportable.title if reportable.respond_to?(:title)
    return reportable.name if reportable.respond_to?(:name)
    return reportable.id
  end
```

Reply:
Done